fuse/namespace.cpp

```diff
@@ -301,44 +296,35 @@ bool get_workspace_dir(
 	return false;
 }
 
-bool setup_user_namespaces(const JAST& jast) {
+bool setup_user_namespaces(
+	int id_user,
+	int id_group,
+	bool isolate_network,
+	std::string hostname,
+	std::string domainname)
+{
 	uid_t real_euid = geteuid();
 	gid_t real_egid = getegid();
 
-	uid_t euid = real_euid;
-	gid_t egid = real_egid;
-
-	const std::string id_user = jast.get("user-id").value;
-	if (!id_user.empty())
-		euid = std::stoi(id_user);
+	uid_t euid = id_user;
+	gid_t egid = id_group;
 
-	const std::string id_group = jast.get("group-id").value;
-	if (!id_group.empty())
-		egid = std::stoi(id_group);
+	int flags = CLONE_NEWNS|CLONE_NEWUSER|CLONE_NEWUTS;
 
-	int flags = CLONE_NEWNS|CLONE_NEWUSER;
-
-	for (auto &res : jast.get("resources").children) {
-		const std::string &key = res.second.value;
-		if (key == "isolate/host") flags |= CLONE_NEWUTS;
-		if (key == "isolate/net") flags |= CLONE_NEWNET;
-	}
+	if (isolate_network)
+		flags |= CLONE_NEWNET;
 
 	// Enter a new mount namespace we can control
 	if (0 != unshare(flags)) {
 		std::cerr << "unshare: " << strerror(errno) << std::endl;
 		return false;
 	}
 
-	// Wipe out our hostname
-	if (0 != (flags & CLONE_NEWUTS)) {
-		if (sethostname("build", 5) != 0) {
-			std::cerr << "sethostname(build): " << strerror(errno) << std::endl;
-		}
-		if (setdomainname("local", 5) != 0) {
-			std::cerr << "setdomainname(local): " << strerror(errno) << std::endl;
-		}
-	}
+	if (sethostname(hostname.c_str(), hostname.length()) != 0)
+		std::cerr << "sethostname(" << hostname << "): " << strerror(errno) << std::endl;
+
+	if (setdomainname(domainname.c_str(), domainname.length()) != 0)
+		std::cerr << "setdomainname(" << domainname << "): " << strerror(errno) << std::endl;
```

Comment:
Must allow "real" hostname to be unmodified, since this can break networking access for some tools.